### PR TITLE
Enforce Unique Token/Service ConnectedAccount Combo

### PR DESCRIPTION
**Pull request by kewitz**
Resolves https://github.com/opencollective/opencollective/issues/2957

### CODE REVIEW

File: server/models/ConnectedAccount.js [hunk 1]
```diff
@@ -43,6 +43,13 @@ export default (Sequelize, DataTypes) => {
     {
       paranoid: true,
 
+      indexes: [
```
Comment (Betree) [inline server/models/ConnectedAccount.js]: Is this enough to create the index in DB, no need to create a migration?

Comment (kewitz) [reply server/models/ConnectedAccount.js]: Nice catch @Betree! It is not needed if you use `db.sync()`, since we don't, I'll add it to the migration.

**Comment by kewitz** [comment]
Rebased.

**Comment by kewitz** [comment]
@znarf @Betree: what should we do with this migration? Are we OK with restricting this through the DB or should I manually validate this through the GraphQL API for TransferWise only?

**Comment by Betree** [comment]
I am. DB constraints are always better, they make sure that the data stays consistent no matter what we do in the code.

Are there any cases or existing data that doesn't play nice with it?

**Comment by znarf** [comment]
I'm just unsure about the deletion policy. For example, we need to protect Twitter connectedAccounts that are actively used and configured for automated tweets.

**Comment by kewitz** [comment]
I agree with @znarf, this is a legit use for duplicated tokens. I'll implement this in the mutator level for now.

**Comment by znarf** [comment]
@kewitz we still want that, need to go extra mile and figure a sensible deletion policy.

**Comment by kewitz** [comment]
Cool, I'll generate a report and properly analyze all cases this week.

**Comment by znarf** [comment]
@kewitz I'm gonna work on that.

**Comment by kewitz** [comment]
Thanks @znarf!

**Comment by znarf** [comment]
I'm almost done for Twitter, main strategy is:
- Ignore deleted ConnectedAccounts and Collectives
- Delete duplicates that are not Collective or User=isHost or Organization=isHost (there is no point connecting Twitter out of these cases)
- Delete duplicates that don't have Tweets configured or are explicitly turned off
- Deletion of the least used/useful (no transaction, no tweet going out)

With this strategy, I'm done to 1 duplicate.

**Comment by znarf** [comment]
Stale, closing.

**Pull request closed without merging**